### ISSUE TITLE
Please consider switching from ron to toml for config files

### ISSUE THREAD

**Issue by wiiznokes** [pop-os/cosmic-epoch#271]
Config files in COSMIC are plain text so that users can edit them manually.

I think we should consider a more user friendly language for it.

toml has a stable syntax that is well documented. Famous software already use it for configuration like Rust, Helix or Hyprland(at least a little i think).

advantages:

- popular / great tooling / rust support
- nested structure scale well horizontally (imo)
- people know toml in general

In this [reddit post](https://www.reddit.com/r/rust/comments/14qd6dx/what_config_format_do_you_prefer/), you can see that Toml wins:

```
1. Toml: 1400
2. Json: 795
3. Yaml: 774
...
6 and last. Ron: 76
```

**Comment by wiiznokes** [comment]
And also, please consider to not splitting to much config into multiple files, i don't think it's worth for setting that don't change a lot. It help if you want for example read a config from another person, copy paste a config, or just saving that file.

I can't think of a setting in COSMIC that change so often that it need to be separate.

**Comment by soloturn** [comment]
toml? oh my god :) there are much nicer options than toml if one really does not want ron.

* cuddle: https://kdl.dev/
* dhall: https://dhall-lang.org/

**Comment by wiiznokes** [comment]
Cuddle seems ok to me, but it has no support for serde yet so idt it's a possible solution for now.

dhall seems way to complex for COSMIC. Imo, it could be used in particular case, for complex plugin for example, but it shouldn't be used for general configuration management.

I think toml is a good solution for now. It have battle tested rust implementation.
And the main complain i saw on internet about toml is that is doesn't fit well for nested struct. But it should be fixed in toml 1.1.0, with support for multi-line table.

Anyway, it's not that hard to change the format in the future, so imo, we should just take the best available solution that currently exist. And that's clearly not ron.

**Comment by atagen** [comment]
> we should just take the best available solution that currently exist. And that's clearly not ron.

why? if the intent would be for users to engage the settings UI, human readable config is a niche issue. it's not clear at all what the issue with RON is aside from unfamiliarity.

**Comment by wiiznokes** [comment]
`.kdl` have 2 derives implementation, and niri use this format.

**Comment by nevdelap** [comment]
> > we should just take the best available solution that currently exist. And that's clearly not ron.
> 
> why? if the intent would be for users to engage the settings UI, human readable config is a niche issue. even then it's not clear at all what the issue with RON is aside from unfamiliarity.

I agree with those who think that human readable config is NOT at all a niche issue. It is a standard way for advanced users to interact with things in Linux.

But now that I know that the format is ron and I've installed the right extension to support ron in VSCode, I'm happy enough with it, though if it changed to something that I use more often, like toml, I'd be happy with that too.

**Comment by nevdelap** [comment]
> It is a standard way for advanced users to interact with things in Linux.

I wasn't referring to compositors or DE's, I was referring to the philosophy of Linux in general vs 'human readable config = niche issue'. I'm happy editing the ron.
